lib/ruby_lsp/requests/code_action_resolve.rb

```diff
@@ -135,6 +149,63 @@ def perform
         )
       end
 
+      sig { returns(T.any(Interface::CodeAction, Error)) }
+      def refactor_method
+        return Error::EmptySelection if @document.source.empty?
+
+        source_range = @code_action.dig(:data, :range)
+        return Error::EmptySelection if source_range[:start] == source_range[:end]
+
+        scanner = @document.create_scanner
+        start_index = scanner.find_char_position(source_range[:start])
+        end_index = scanner.find_char_position(source_range[:end])
+        extracted_source = T.must(@document.source[start_index...end_index])
+
+        # Find the closest statements node, so that we place the refactor in a valid position
+        closest_statements, _ = T.cast(
```

Comment:
```suggestion
        # Find the closest method declaration node, so that we place the refactor in a valid position
        closest_def, _ = T.cast(
```